cmd/cli/cmd/root_test.go

```diff
@@ -0,0 +1,284 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	net2 "net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/AccumulateNetwork/accumulate/config"
+	"github.com/AccumulateNetwork/accumulate/internal/api"
+	api2 "github.com/AccumulateNetwork/accumulate/internal/api/v2"
+	"github.com/AccumulateNetwork/accumulate/internal/logging"
+	"github.com/AccumulateNetwork/accumulate/internal/node"
+	"github.com/AccumulateNetwork/accumulate/internal/relay"
+	acctesting "github.com/AccumulateNetwork/accumulate/internal/testing"
+	"github.com/AccumulateNetwork/accumulate/networks"
+	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/net"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+	"github.com/tendermint/tendermint/rpc/client/local"
+)
+
+type testCase func(t *testing.T, tc *testCmd)
+type testMatrixTests []testCase
+
+var testMatrix testMatrixTests
+
+func TestCli(t *testing.T) {
+	tc := &testCmd{}
+	tc.initalize(t)
+	defer tc.finalize(t)
+
+	//set mnemonic for predictable addresses
+	_, err := tc.execute(t, "key import mnemonic yellow yellow yellow yellow yellow yellow yellow yellow yellow yellow yellow yellow")
+	require.NoError(t, err)
+
+	testMatrix.execute(t, tc)
+}
+
+func GetFunctionName(i interface{}) string {
+	return runtime.FuncForPC(reflect.ValueOf(i).Pointer()).Name()
+}
+
+func (tm *testMatrixTests) addTest(tc testCase) {
+	*tm = append(*tm, tc)
+
+}
+
+func (tm *testMatrixTests) execute(t *testing.T, tc *testCmd) {
+
+	// Sort by testCase number
+	sort.SliceStable(*tm, func(i, j int) bool {
+		return GetFunctionName((*tm)[i]) < GetFunctionName((*tm)[j])
+	})
+
+	//execute the tests
+	for _, f := range testMatrix {
+		t.Log(GetFunctionName(f))
+		f(t, tc)
+	}
+}
+
+type testCmd struct {
+	rootCmd        *cobra.Command
+	directoryCmd   *exec.Cmd
+	validatorCmd   *exec.Cmd
+	defaultWorkDir string
+	restPort       int
+	jsonRpcPort    int
+}
+
+func (c *testCmd) finalize(t *testing.T) {
+	//c.validatorCmd.Process.Kill()
+	//c.directoryCmd.Process.Kill()
+
+	//os.RemoveAll(c.defaultWorkDir)
+}
+
+//NewTestBVNN creates a BVN test Node and returns the rest and jsonrpc ports
+func NewTestBVNN(t *testing.T, defaultWorkDir string) (int, int) {
+	t.Helper()
+	opts, err := acctesting.NodeInitOptsForNetwork(acctesting.LocalBVN)
+	require.NoError(t, err)
+	opts.WorkDir = defaultWorkDir
+	opts.Port, err = net.GetFreePort()
+	require.NoError(t, err)
+	cfg := opts.Config[0]
+	cfg.Mempool.MaxBatchBytes = 1048576
+	cfg.Mempool.CacheSize = 1048576
+	cfg.Mempool.Size = 50000
+	cfg.Accumulate.API.EnableSubscribeTX = true
```

Comment:
For these tests, I think we should leave `EnableSubscribeTX` disabled. Test setup enables this because some tests use it. But we don't use it in production, so I think we should leave it disabled for test matrix tests.